src/commands/external.rs

```diff
@@ -13,11 +15,30 @@ pub async fn execute_external_subcommand(args: Vec<String>) -> Result<()> {
     let plugin_name = args.first().ok_or_else(|| anyhow!("Expected subcommand"))?;
     let plugins_dir = get_spin_plugins_directory()?;
     check_plugin_spin_compatibility(plugin_name, env!("VERGEN_BUILD_SEMVER"), &plugins_dir)?;
-    let path = plugins_dir.join(plugin_name).join(plugin_name);
-    let mut command = Command::new(path);
+    let path = plugins_dir.join(plugin_name);
+    let binary = path.join(plugin_name);
+    let mut command = Command::new(binary);
     if args.len() > 1 {
         command.args(&args[1..]);
     }
+    let mut env_vars: HashMap<String, String> = HashMap::new();
+    // Add version of spin
+    env_vars.insert(
+        "SPIN_VERSION".to_string(),
+        env!("VERGEN_BUILD_SEMVER").to_owned(),
+    );
+    // Add path to spin
+    let spin_bin_path = env::current_exe()?.to_str().unwrap_or_default().to_owned();
+    env_vars.insert("SPIN_BIN_PATH".to_string(), spin_bin_path);
+    // Add version of spin
+    let spin_cwd = env::current_dir()?.to_str().unwrap_or_default().to_owned();
+    env_vars.insert("SPIN_CWD".to_string(), spin_cwd);
```

Comment:
instead of multiple inserts, wonder if it would be neater to do something like the following --- the more i think about it, this is probably unnecessary but fun to point out:
```
 let map: HashMap<String, String> = vec![
        ('SPIN_VERSION', "val"),
        ('SPIN_BIN_PATH', "val"),
        ('SPIN_CWD', "val"),
    ].into_iter().collect();
```